### ISSUE TITLE
error importing pyqtgraph in macosx

### ISSUE THREAD

**Issue by amrit-poudel** [pyqtgraph/pyqtgraph#285]
I get the following error when I try to import pyqtgraph in macosx 10.2, python version 2.7.9

> > > import pyqtgraph.examples
> > > pyqtgraph.examples.run()
> > > This application failed to start because it could not find or load the Qt platform plugin "cocoa".

Available platform plugins are: cocoa, minimal, offscreen.

Reinstalling the application may fix this problem.

I have QT library, PyQt4, numpy/scipy/matplotlib installed on my system and PATH, LD_LIBRARY_PATH and PYTHONPATH are all set correctly.

Surprisingly, Qt4Agg package works fine under matplotlib which also depends on PyQt4.

 I am not sure why I get this error in case of pyqtgraph.

**Comment by campagnola** [comment]
Sounds like a PyQt installation issue. Perhaps you have multiple versions installed? Try just this:

    from PyQt4 import QtGui
    app = QtGui.QApplication([])

**Comment by amrit-poudel** [comment]
I tried importing PyQt4 like you suggested, and it seems to work fine.

**Comment by campagnola** [comment]
Ok, so what happens when you try:

    import PyQt4
    import pyqtgraph

**Comment by amrit-poudel** [comment]
I tried that and it does not throw any error.

**Comment by campagnola** [comment]
Ok, getting closer. I think you just need to figure out which PyQt (or PySide) module is being loaded by pyqtgraph. What happens with just

    import pyqtgraph

?

**Comment by amrit-poudel** [comment]
I do not have PySide installed.  import pyqtgraph also works fine without any error.

**Comment by campagnola** [comment]
You said above "I get the following error when I try to import pyqtgraph". Is that fixed now? You'll need to do some sleuthing here--figure out exactly when the error occurs.

**Comment by amrit-poudel** [comment]
Sorry about the confusion. I was not able to run pyqtgraph examples. I should have stated that clearly.

import pyqtgraph.examples
pyqtgraph.examples.run()

A new window pops up, and when I click on "run example", I get the following erorr:

This application failed to start because it could not find or load the Qt platform plugin "cocoa".

Available platform plugins are: cocoa, minimal, offscreen.

Reinstalling the application may fix this problem.

**Comment by cboulay** [comment]
I'm guessing Spyder is using a different version of python (or at least a different site_packages dir) than terminal's `python`.

From terminal, `which python`. Run `python` and check the version. Run `ipython` and check the version.

From Spyder, check the version of python and ipython from the embedded console.

If they are indeed different versions then there are plenty of tutorials in the wild on how to fix that. Don't delete system python (OS X needs that), but I wouldn't use it for developing or any personal use. For development and personal use I recommend installing python and some of its compiling-needed packages (e.g., scipy, anything that fails to install easily with `pip`) with "homebrew".

Not related to your question, but I used to use Spyder on OS X then I found it got unbearably slow for me. The on-screen text entry was literally seconds behind my fingers. I recently switched to PyCharm CE and it works great. I use it with pyqtgraph with no problem, even when debugging flowcharts. Though, I was using homebrew's spyder, I didn't build from source, so the problem may have been fixed in whatever version you're using.

**Comment by amrit-poudel** [comment]
I built spyder from source(including all the dependencies, all from the source) and am not using an app. As such, python (v 2.7.9, also built from source) on the terminal or on the spyder are the same. 
Spyder works fine. The only problem is I cannot run pyqtgraph examples. 
Will appreciate any other suggestions. Thanks.

**Comment by cboulay** [comment]
Everything you describe points to spyder's python having PyQt4 available but terminal's python does not. The reason the examples don't work is because they're spawning independent python processes that are not using spyder's environment. If you're certain they're both running the same python binaries, then maybe spyder or other pytthon is using a virtualenv.

**Comment by amrit-poudel** [comment]
I am sorry about the confusion, but there are not two python anywhere. Like I said before, I built spyder from source using python 2.7.9, and I use the same python in spyder IDE or in the terminal. I do have PyQt4 installed. There is only one PyQt4 and spyder uses the same PyQt4 as the python in the terminal.

I get this error when I try to import pyqt4 in the terminal:

>>> from PyQt4 import QtGui
>>> app = QtGui.QApplication([])
This application failed to start because it could not find or load the Qt platform plugin "cocoa".

Available platform plugins are: cocoa, minimal, offscreen.

Reinstalling the application may fix this problem.

Abort trap: 6

I get the same error when I try to import pyqtgraph.examples :

import pyqtgraph.examples
pyqtgraph.examples.run()
This application failed to start because it could not find or load the Qt platform plugin "cocoa".

**Comment by cboulay** [comment]
I don't know what's going on with your system, sorry. I have pyqtgraph working fine on my system but I used homebrew to install python and pyqt. If that's an option for you then I would suggest trying that. This is probably one of those problems that is not worth the hours required to find out that some makefile somewhere made an assumption about a folder structure that is no longer valid.

**Comment by amrit-poudel** [comment]
My only confusion is that Qt4Agg backend works fine from spyder, but pyqtgraph does not.

Like pyqtgraph, Qt4Agg also requires PyQt4 installed in the system.  This means my PyQt4 installation is not all bad, otherwise it would not have worked with Qt4Agg backend in matplotlib either.

I am just confused why pyqtgraph would not work from spyder. I do not expect it to work from the terminal due to the error I mentioned above, but it should work from spyder just like Qt4Agg.

Thanks for your input. I appreciate it.

**Comment by 2xB** [comment]
As no effort to reproduce this was successfull, I will close this issue now. If it still persists or there is new input to this, feel free to leave a comment or open a new pull request.
